plugins/router/template/router.go

```diff
@@ -144,6 +145,20 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 	return router, nil
 }
 
+func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
+	if len(alias.PreferPort) == 0 {
+		return svc.EndpointTable
+	}
+	endpoints := make([]Endpoint, 0, len(svc.EndpointTable))
+	for i := range svc.EndpointTable {
+		endpoint := svc.EndpointTable[i]
+		if endpoint.PortName == alias.PreferPort || endpoint.Port == alias.PreferPort {
```

Comment:
I think I'm missing something here.  How is the use case of specifying multiple ports in my route api object supported by our template router?  What I see is:

1.  The endpoints are created from all endpoints in the endpoints object (one more "endpoints" for good measure) but filtering anything that is UDP if configured to do so
2.  A preferred port is always set for specs with ports > 0
3.  The template in our default impl will call this method which will only include the preferred port which is either a single name or port number

I'm not seeing how my Route.Spec.Ports settings filter down to the template to select a subset of the endpoint table.  Perhaps this is the IP part of the WIP.